### PR TITLE
Log

### PR DESCRIPTION
**Pull request by berengerwss**
I've added these features to the `frm::ui::Log` class :
1. added an m_`elapsedTimeSinceLastMessage `private member with its …
2. added an methods to clear messages.
3. added an `ImGuiFilter `.
4. indentation harmonized.

### CODE REVIEW

File: src/all/frm/ui/Log.cpp [hunk 1]
```diff
@@ -146,4 +147,45 @@ void Log::draw()
...
+	m_msgBufNext = 0;
+	m_txtBuf     = 0;
+	m_txtBufNext = 0;
+	m_lastLog    = 0;
+	m_lastDbg    = 0;
+	m_lastErr    = 0;
+
+	/* Create new empty buffers */
+	createBuffers();
+}
```
Comment (john-chapman) [inline src/all/frm/ui/Log.cpp]: Deleting the buffers isn't necessary, just reset the 'head' and 'next' ptrs.

**Comment by john-chapman** [comment]
I can't merge this because of the changes to .gitattributes (your fork removes git LFS and most of the data directory). You'll need to make a new branch on your fork, checkout upstream and then cherry-pick the commits you want to PR:

https://stackoverflow.com/questions/34027850/how-to-pull-request-a-specific-commit

File: src/all/frm/ui/Log.cpp (hunk 1 above)
Comment (berengerwss) [reply src/all/frm/ui/Log.cpp]: I tried to only reset m_msgBufHead and m_msgBufNext and it crashes in Log.cpp line 92 because m_msgBufNext is NULL.

Comment (john-chapman) [reply src/all/frm/ui/Log.cpp]: Don't set them to 0, set them to point at the msg/txt buffers as per the ctor.

**Pull request closed without merging**